src/server/src/main/java/io/cassandrareaper/storage/CassandraStorage.java

```diff
@@ -559,20 +559,6 @@ private boolean addClusterAssertions(Cluster cluster) {
         "Cluster should not be persisted with UNKNOWN state");
 
     Preconditions.checkState(cluster.getPartitioner().isPresent(), "Cannot store cluster with no partitioner.");
-    // assert we're not overwriting a cluster with the same name but different node list
-    Set<String> previousNodes;
-    try {
-      previousNodes = getCluster(cluster.getName()).getSeedHosts();
-    } catch (IllegalArgumentException ignore) {
-      // there is no previous cluster with same name
-      previousNodes = cluster.getSeedHosts();
-    }
-    Set<String> addedNodes = cluster.getSeedHosts();
-
-    Preconditions.checkArgument(
-        !Collections.disjoint(previousNodes, addedNodes),
-        "Trying to add/update cluster using an existing name: %s. No nodes overlap between %s and %s",
-        cluster.getName(), StringUtils.join(previousNodes, ','), StringUtils.join(addedNodes, ','));
```

Comment:
isn't this condition checking something else: it checks that the cluster being added doesn't already exist. 

Is this dangerous? Accidenty adding a new cluster that re-uses another clusters name will wipe it.
I rather require the user to delete (unregister) the previous cluster first, especially now that force delete is available.

Reply:
but if we keep the check, it's impossible to renew the seed list of a cluster that changed all of its ips, which is what we're trying to do.
How would you change the check to allow this?

Unregistering the cluster would remove the schedules and the history, which we're trying to avoid.

Reply:
> How would you change the check to allow this?

We don't.
It's important that it remains impossible to replace an existing cluster with a new cluster where none of the ipaddresses overlap.

If an existing cluster changes the ipaddress of all its nodes, in a timeframe short enough that Reaper did not notice, then the cluster needs be unregistered before it can be added to Reaper again.

There is a caveat here: when the cluster has become stale and moves into INACTIVE state. Then it might be an option to make it possible, but i'm still uncomfortable with making this too easy. I would suggest a yaml setting that permits to overriding INACTIVE clusters, which defaults to the safe value of false.

Reply:
>If an existing cluster changes the ipaddress of all its nodes, in a timeframe short enough that Reaper did not notice, then the cluster needs be unregistered before it can be added to Reaper again. We know from experience that relying upon the cluster-name as an identifier is dangerous. The set of ipaddresses is a better identifier to the cluster.

I would like to suggest avoid relying on IP addresses as a cluster identifier. It will not work clusters running in Kubernetes. If I have a cluster in Kubernetes and do a rolling restart of the cluster, then every  Cassandra node will wind up with a new IP address.

What about having a UID for each cluster? It could either provided by users/clients or generated by Reaper.

Reply:
Such an UUID would have to be provided by Cassandra itself. We could rely on the Host-IDs instead of the ip-addresses. I believe the Host-IDs stay the same even when you do a `"-Dcassandra.replace_address=…"`